resources/repo.rb

```diff
@@ -0,0 +1,88 @@
+property :repo_name, String, name_attribute: true
+property :repo_type, String, default: 'maven-hosted'
+property :config, Hash
+property :api_url, String
+property :api_user, String
+property :api_password, String
+
+def apiclient
+  @apiclient ||= Nexus3::Api.new(api_url, api_user, api_password)
+end
+
+def textify(args)
+  require 'json'
+  JSON.generate(args)
+end
+
+load_current_value do
+  apiclient.list_repositories
+end
+
+action :create do
+  nexus_api 'get_repo' do
+    action :create
+    endpoint new_resource.api_url
+    username new_resource.api_user
+    password new_resource.api_password
+
+    content <<EOS
+    import groovy.json.JsonSlurper
+    import groovy.json.JsonOutput
+
+    def params = new JsonSlurper().parseText(args)
+
+    JsonOutput.toJson(repository.repositoryManager.get(params.name).getConfiguration())
+EOS
+  end
+
+  converge_if_changed do
+    nexus_api 'create_or_update_repo' do
+      # TODO: Current nexus3_api passes a string and text/plain, change this back to a Hash later
+      args textify(name: new_resource.repo_name, type: new_resource.repo_type)
+      action :create, :run
+      endpoint new_resource.api_url
+      username new_resource.api_user
+      password new_resource.api_password
+
+      content <<EOS
+    // TODO: document parameters passed to this script.
+    // name: repo name
+    // type: repo type (equal to recipe for now?)
+    import groovy.json.JsonSlurper
+    import groovy.json.JsonOutput
+
+    def params = new JsonSlurper().parseText(args)
+    // TODO: add checks for missing parameters.
+
+    if (repository.repositoryManager.get(params.name) == nil) {
+        // create
+        def config = new Configuration(
+            repositoryName: params.name,
+            recipeName: params.type
+        )
+        repository.repositoryManager.createRepository(config)
+    } else {
+        // update
+        config = repository.repositoryManager.get(params.name).getConfiguration()
```

Comment:
OPT: You could use the groovy safe operator to perform only 1 repo retrival `repository.repositoryManager.get(params.name)?.getConfiguration()`